hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestDynamoDBMetadataStore.java

```diff
@@ -274,7 +300,9 @@ public void testInitialize() throws IOException {
         getTestTableName("testInitialize");
     final Configuration conf = s3afs.getConf();
     conf.set(S3GUARD_DDB_TABLE_NAME_KEY, tableName);
-    try (DynamoDBMetadataStore ddbms = new DynamoDBMetadataStore()) {
+    enableOnDemand(conf);
```

Comment:
we could add this to @Before, right?

Reply:
it's done in the beforeClass for the static table; for the others its done to to the config which comes from the FS.
I'm going to change it to set it in the FS config before we set the FS. If someone has per-bucket options to set the capacity then things will be different, but I'd be surprised if anyone ever does that